### PR TITLE
update all tests to use quicktest

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -62,11 +56,11 @@ var callTests = []struct {
 		Body: struct{ I bool }{true},
 	},
 	expectError: `Post http:.*: cannot unmarshal parameters: cannot unmarshal into field Body: cannot unmarshal request body: json: cannot unmarshal .*`,
-	assertError: func(c *gc.C, err error) {
-		c.Assert(errgo.Cause(err), gc.FitsTypeOf, (*httprequest.RemoteError)(nil))
-		err1 := errgo.Cause(err).(*httprequest.RemoteError)
-		c.Assert(err1.Code, gc.Equals, "bad request")
-		c.Assert(err1.Message, gc.Matches, `cannot unmarshal parameters: cannot unmarshal into field Body: cannot unmarshal request body: json: cannot unmarshal .*`)
+	assertError: func(c *qt.C, err error) {
+		err1, ok := errgo.Cause(err).(*httprequest.RemoteError)
```
Comment (rogpeppe) [inline client_test.go]: I wonder if we should put FitsTypeOf into quicktest too. You might consider writing a local FitsTypeOf checker just to make the diff smaller, I guess.

File: fancyerror_test.go [hunk 2]
```diff
@@ -420,16 +417,20 @@ YUI().use('storefront-cookie', 'storefront-utils', 'user-dropdown',
...
+	expectError: `unexpected content type text/html; want application/json; content: Page not found \| Juju; Jump to content; Store; Demo; About; Features; Community; Docs; Get started; ☰; Create; \+; 404: Sorry, we couldn’t find the page; Try a different URL, try searching for solutions or learn how to; create your own solution; Browse the store; All bundles; All charms; Submit a bug; Browse the store ›; Back to the top; Demo; About; Features; Docs; Get Started; Juju on Google\+; Ubuntu Cloud on Twitter; Ubuntu Cloud on Facebook; © 2015 Canonical Ltd. Ubuntu and Canonical are registered trademarks of Canonical Ltd; Legal information; Report a bug on this site; Got to the top of the page`,
```
Comment (rogpeppe) [inline fancyerror_test.go:420]: Thanks for fixing the quotes here. I'm surprised it matched before.

File: handler_test.go [hunk 3]
```diff
@@ -466,35 +466,38 @@ var handlePanicTests = []struct {
...
-	for i, test := range handlePanicTests {
-		c.Logf("%d: %s", i, test.expect)
-		c.Check(func() {
-			testServer.Handle(test.f)
-		}, gc.PanicMatches, test.expect)
+func TestHandlePanicsWithBadFunctions(t *testing.T) {
+	c := qt.New(t)
+
+	for _, test := range handlePanicTests {
+		c.Run(test.expect, func(c *qt.C) {
```
Comment (rogpeppe) [inline handler_test.go]: using "expect" as the test name here is going to lead to some mighty awkward test names. Perhaps consider adding a testName field and using that?

File: handler_test.go [hunk 4]
```diff
@@ -1233,65 +1278,67 @@ func (s *handlerSuite) TestHandleErrorsWithErrorAfterWriteHeader(c *gc.C) {
...
-	c.Assert(req1.Host, gc.Equals, req2.Host)
-	c.Assert(req1.Form, jc.DeepEquals, req2.Form)
-	c.Assert(req1.PostForm, jc.DeepEquals, req2.PostForm)
-	c.Assert(req1.MultipartForm, jc.DeepEquals, req2.MultipartForm)
-	c.Assert(req1.Trailer, jc.DeepEquals, req2.Trailer)
-	c.Assert(req1.RemoteAddr, gc.Equals, req2.RemoteAddr)
-	c.Assert(req1.RequestURI, gc.Equals, req2.RequestURI)
-	c.Assert(req1.TLS, gc.Equals, req2.TLS)
-	c.Assert(req1.Cancel, gc.Equals, req2.Cancel)
+func assertRequestEquals(c *qt.C, req1, req2 *http.Request) {
```
Comment (rogpeppe) [inline handler_test.go]: This could probably be done less awkwardly now with cmpopts. Maybe something for later - perhaps leave a TODO?

**Review by rogpeppe — APPROVED** [review]
LGTM with a few minor comments and suggestions, thanks.

File: handler_test.go (hunk 3 above)
Comment (mhilton) [reply handler_test.go]: done

File: handler_test.go (hunk 4 above)
Comment (mhilton) [reply handler_test.go]: done

File: client_test.go (hunk 1 above)
Comment (mhilton) [reply client_test.go]: I did this with a qt.Satisfies and a predicate function, but that still makes the diff smaller.

**Review by rogpeppe — APPROVED** [review]
LGTM thanks!

**Pull request merged**